### PR TITLE
Add Rubocop support

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+Layout/LineLength:
+  Exclude:
+    - "test/exercises/*"
+    - "lib/analyzers/*/analyze.rb"
+
+Lint/SuppressedException:
+  Exclude:
+    - "test/**/*"
+    - "lib/analyzers/exercise_analyzer.rb"
```
Comment (ErikSchierboom) [inline .rubocop.yml:35]: This was added to ignore the linting error for [this line](https://github.com/exercism/ruby-analyzer/blob/master/lib/analyzers/exercise_analyzer.rb#L20). Let me know if that is okay

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Layout/MultilineMethodCallIndentation:
+  EnforcedStyle: indented
+
+Layout/EmptyLinesAroundAccessModifier:
+  EnforcedStyle: only_before
+
+Layout/LineLength:
+  Exclude:
+    - "test/exercises/*"
+    - "lib/analyzers/*/analyze.rb"
```
Comment (ErikSchierboom) [inline .rubocop.yml:30]: Some of the analyzers have quite long line lengths (see [this line as an example](https://github.com/exercism/ruby-analyzer/blob/master/lib/analyzers/two_fer/analyze.rb#L12)). I quite like the current formatting, even though the lines are long. Let me know if you agree with this exclusion or would prefer to fix the line length.

File: .rubocop.yml (hunk 1 above)
Comment (iHiD) [reply .rubocop.yml:35]: Yep, that's fine.

File: .rubocop.yml (hunk 2 above)
Comment (iHiD) [reply .rubocop.yml:30]: Yep - this is good too. Thanks! :)

**Comment by iHiD** [comment]
Needs rebasing.

File: lib/analyzers/two_fer/analyze.rb [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
-    no_method:               "ruby.general.no_target_method",
-    incorrect_indentation:   "ruby.general.incorrect_indentation",
-    explicit_return:         "ruby.general.explicit_return",         # "The last line automatically gets returned"
-    splat_args:              "ruby.two-fer.splat_args",              # "Rather than using *%s, how about actually setting a parameter called 'name'?"
-    missing_default_param:   "ruby.two-fer.missing_default_param",   # "There is no correct default param - the tests will fail"
+    no_module: "ruby.general.no_target_module",
+    no_method: "ruby.general.no_target_method",
+    incorrect_indentation: "ruby.general.incorrect_indentation",
+    explicit_return: "ruby.general.explicit_return", # "The last line automatically gets returned"
+    splat_args: "ruby.two-fer.splat_args", # "Rather than using *%s, how about actually setting a parameter called 'name'?"
```
Comment (kotp) [inline lib/analyzers/two_fer/analyze.rb:7]: Not a request to change it in this PR but "how'bout actually" seems a bit less ideal than "what if we set a argument called "name"?

File: .rubocop.yml (hunk 1 above)
Comment (kotp) [reply .rubocop.yml:35]: That line should probably not be contained within the `begin` and `end` block, as that is provided by the method definition.

```ruby
  def call
    analyze!
    rescue FinishedFlowControlException
    results
  end```

**Pull request merged**